ssl/record/ssl3_record.c

```diff
@@ -271,8 +271,13 @@ int ssl3_get_record(SSL *s)
                 thisrr->type = type;
                 thisrr->rec_version = version;
 
-                /* Lets check version. In TLSv1.3 we ignore this field */
+                /*
+                 * Lets check version. In TLSv1.3 we ignore this field. For an
+                 * HRR we haven't actually selected TLSv1.3 yet, but we still
+                 * treat it as TLSv1.3, so we must check for that explicitly
+                 */
                 if (!s->first_packet && !SSL_IS_TLS13(s)
+                        && !s->hello_retry_request
```

Comment:
For the actual HelloRetryRequest record, ```s->first_packet``` should be set, right?
So the effect of checking ```s->hello_retry_request``` is really for when we are processing the ServerHello, which is in this flow the second record we're reading?

Reply:
Right! Good point. I've tweaked the comment wording accordingly.